### PR TITLE
MAISTRA-2057 Add missing _affinity.tpl to the istio-control helm templates

### PR DESCRIPTION
**Pull request by shamsher31**

### CODE REVIEW

**Comment by dgn** [comment]
Looks like a legit failure: 

```
Error: Deployment.apps "istiod" is invalid: spec.template.spec.affinity.nodeAffinity.requiredDuringSchedulingIgnoredDuringExecution.nodeSelectorTerms[0].matchExpressions[0].values: Required value: must be specified when `operator` is 'In' or 'NotIn'
```

File: manifests/charts/istio-control/istio-discovery/values.yaml [hunk 1]
```diff
@@ -199,6 +222,14 @@ global:
...
     #   memory: 128Mi
 
+  # Default node tolerations to be applied to all deployments so that all pods can be
+  # scheduled to a particular nodes with matching taints. Each component can overwrite
+  # these default values by adding its tolerations block in the relevant section below
+  # and setting the desired values.
+  # Configure this field in case that all pods of Istio control plane are expected to
+  # be scheduled to particular nodes with specified taints.
+  defaultTolerations: []
+
```
Comment (dgn) [inline manifests/charts/istio-control/istio-discovery/values.yaml:232]: I think you'll also need to add `global.arch`?

Comment (shamsher31) [reply manifests/charts/istio-control/istio-discovery/values.yaml:232]: Thanks @dgn

**Review by jwendell — APPROVED** [review]

**Review by luksa — APPROVED** [review]

**Review by rcernich — APPROVED** [review]

**Pull request merged**